matchbox/resource_profile.go

```diff
@@ -112,12 +139,13 @@ func resourceProfileCreate(d *schema.ResourceData, meta interface{}) error {
 func validateResourceProfile(d *schema.ResourceData) error {
 	_, hasRAW := d.GetOk("raw_ignition")
 	_, hasCLC := d.GetOk("container_linux_config")
-	if hasCLC && hasRAW {
-		return errors.New("container_linux_config and raw_ignition are mutually exclusive")
+	_, hasGEN := d.GetOk("generic")
+	if (hasCLC && hasRAW) || (hasCLC && hasGEN) || (hasRAW && hasGEN) {
+		return errors.New("container_linux_config, raw_ignition and generic are mutually exclusive")
 	}
 
-	if !hasCLC && !hasRAW {
-		return errors.New("container_linux_config or raw_ignition are required")
+	if !hasCLC && !hasRAW && !hasGEN {
```

Comment:
Seems like we might want to drop all these requirements. In the simplest case, a user might only want to serve iPXE configs (kernel, initrd, and args).